src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/FormValueProviderFactory.cs

```diff
@@ -4,35 +4,28 @@
 using System;
 using System.Globalization;
 using System.Threading.Tasks;
-using Microsoft.AspNetCore.Http;
-using Microsoft.AspNetCore.Mvc.Internal;
 
 namespace Microsoft.AspNetCore.Mvc.ModelBinding
 {
     public class FormValueProviderFactory : IValueProviderFactory
     {
-        public Task<IValueProvider> GetValueProviderAsync(ActionContext context)
+        public async Task CreateValueProviderAsync(ValueProviderFactoryContext context)
```

Comment:
This is an important optimization. This code needs to avoid the allocating the task when the body isn't form data. Verify this with the profiler if you want to change the factoring of these methods